ddtrace/_trace/trace_handlers.py

```diff
@@ -777,6 +780,29 @@ def _on_redis_command_post(ctx: core.ExecutionContext, rowcount):
         ctx.span.set_metric(db.ROWCOUNT, rowcount)
 
 
+def _on_redis_execute_pipeline(ctx: core.ExecutionContext, pin, config_integration, args, instance, query):
```

Comment:
any reason why we cannot compute these tags and send into the `core.context_with_data` call and then we can make this data available to anyone listening on this event, I can't see why any listener wouldn't want to have all this data (guessing, not actually sure)

this would also save us from needing to dispatch another event immediately after the context started one.


benefit of the current approach is we only do this processing if we have a listener for that event... but I wonder if we should have a `core.has_listener(event_name)` and then we can use that to skip processing when no one is listening, kind of like `log.isEnabledFor(logging.DEBUG)` 🤔 (this is an aside)

Reply:
I moved all of the tag computation back to the redis contrib. This handler still has the metrics logic that hasn't been generalized yet.